src/eckit/geo/Cache.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+#include <map>
+#include <numeric>
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/geo/util/mutex.h"
+
+
+namespace eckit::geo {
+
+
+class Cache {
+public:
+    using bytes_t = decltype(sizeof(int));
```

Comment:
C++11 section 5.3.3, paragraph 6 (https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3690.pdf):

> The result of sizeof and sizeof... is a constant of type std::size_t. [ Note: std::size_t is defined in the standard header <cstddef> (18.2). — end note ]

So `std::size_t` it is.

Reply:
I could have easily used size_t, but that would be confusing for both the result of a generic cached type footprint and vector<T>, which are here specialised. This is an improvement of similar usage elsewhere (in eckit and below) where this isn't clear, consistent not concise. size_t is generic because it seems widespread usage for, among others, the result of containers size, an index and result of sizeof, neither of which mean the same thing at all.

Please read this way: the type is the result of calling sizeof, which should read clearly that I'm getting the result in bytes (hence the name, but bytes_size_t can be seen as an improvement). So "what is means" not "what it is".

Due to pouplar demand I'll call it bytes_size_t

Reply:
For me it was not so much the naming `bytes_t` vs. `bytes_size_t` vs `size_t` but rather the unnecessary `decltype`.
I would have liked `using bytes_t = std::size_t` as the standard is explicit about the return type of `sizeof()`.
It is just a minor thing which I don't care about so "whatever" :)
Also note that C++17 introduced `std::byte` to represent a single byte. Completely different use but naming is similar.